File: src/main/java/it/unibo/crabinv/Model/Levels/Level.java
package it.unibo.crabinv.Model.Levels;

import it.unibo.crabinv.Model.Enemies.Wave;

public interface Level {
    /**
     * Returns the wave that it is happening
     * @return the current wave
     */
    Wave getCurrentWave();

    /**
     * It's a method that makes the wave advance
     */
    void advanceWave();

    /**
     * Sees if the level is finished if the WaveProvider has not anymore waves
     * @return true if level is finished, false if not
     */
    boolean isLevelFinished();
}

File: src/main/java/it/unibo/crabinv/model/levels/Level.kt
package it.unibo.crabinv.model.levels

import it.unibo.crabinv.model.entities.enemies.wave.Wave

/**
 * Interface that establishes how a level should be and which methods has.
 */
interface Level {
    /**
     * @return the current wave
     */
    val currentWave: Wave?

    /**
     * It's a method that makes the wave advance.
     */
    fun advanceWave()

    /**
     * Sees if the level is finished if the WaveProvider has not anymore waves.
     * @return true if level is finished, false if not
     */
    val isLevelFinished: Boolean
}
